components/pages/home/questAndCollectionTabs.tsx

```diff
@@ -90,52 +85,52 @@ const QuestAndCollectionTabs: FunctionComponent<
   }, [address, quests, trendingQuests]);
 
   const [boosts, setBoosts] = useState<Boost[]>([]);
+  const [completedQuestIds, setCompletedQuestIds] = useState<CompletedQuests>();
+  const [displayBoosts, setDisplayBoosts] = useState<Boost[]>([]);
   const { completedBoostIds } = useContext(QuestsContext);
-  const [claimableQuests, setClaimableQuests] = useState<
-    ClaimableQuestDocument[]
-  >([]);
-  const [pendingBoostClaims, setpendingBoostClaims] = useState<
-    PendingBoostClaim[] | undefined
-  >([]);
 
-  const fetchBoosts = async () => {
+  const fetchBoosts = useCallback(async () => {
+    if (!address) return;
     try {
       const res = await getBoosts();
-      const filteredResponse = res?.filter(
-        (boost) =>
-          (new Date().getTime() - boost.expiry) / MILLISECONDS_PER_WEEK <= 3
-      );
-      if (filteredResponse && filteredResponse?.length > 0)
-        setBoosts(filteredResponse);
+      const completedQuestIdsRes = await getCompletedQuests(address);
+      if (!res || !completedQuestIdsRes) return;
+      if (res?.length === 0 || completedQuestIdsRes?.length === 0) return;
+      setBoosts(res);
+      setCompletedQuestIds(completedQuestIdsRes);
+      const filteredBoosts: Boost[] = [];
+      res?.forEach((boost) => {
+        let userBoostCompletionCheck = true;
+        boost.quests.forEach((quest) => {
+          if (!boost || !completedQuestIds) return;
+          // no quests are completed by user
+          if (!completedQuestIds) return false;
+          // check if all quests are completed by the user and if not then set this flag value to false
+          if (!(completedQuestIdsRes as CompletedQuests).includes(quest))
+            userBoostCompletionCheck = false;
+        });
+        const userBoostCheckStatus = getBoostClaimStatus(address, boost?.id);
+        if (
+          (new Date().getTime() - boost.expiry) / MILLISECONDS_PER_WEEK <= 3 &&
+          boost?.expiry < Date.now() &&
+          userBoostCompletionCheck &&
+          !userBoostCheckStatus &&
+          boost.winner != null
+        ) {
+          filteredBoosts.push(boost);
+        }
+      });
+      if (!filteredBoosts || filteredBoosts.length === 0) return;
+      setDisplayBoosts(filteredBoosts);
     } catch (err) {
       console.log("Error while fetching boosts", err);
     }
```

Comment:
What does it change to use .log or .error ?

Reply:
> Consider using a more robust error handling mechanism instead of logging errors to the console for production environments.
> 
> ```diff
> - console.log("Error while fetching boosts", err);
> + console.error("Error while fetching boosts", err);
> + // Optionally, report the error to an error tracking service
> ```
> 
> Committable suggestion

What does it change to use .log or .error ?

Reply:
Ok I still prefer .log, please change the commit proposition to .log

Reply:
make me a commit proposition on github please